test-cargo-miri/Cargo.toml

```diff
@@ -10,7 +10,7 @@ edition = "2024"
 
 [dependencies]
 byteorder = "1.0"
-cdylib = { path = "cdylib" }
+#cdylib = { path = "cdylib" }
```

Comment:
What's up with this?
We added cdylib dependencies here because we got bug reports about Miri acting strange on them on real code, so we can't just stop supporting that.

Reply:
Dependencies that are both an rlib and cdylib work just fine. However dependencies that are solely a cdylib won't work. But you wouldn't be able to use them anyway. Cargo for some reason insists on still building them, but you would never be able to use them in any other way than copying them from the target dir. It is not an artifact dependency that you can link against through a build script.

Reply:
That looks deliberate: https://github.com/rust-lang/miri/commit/1923044705b2e9d54f7fa8f40e1c0bab5ea081ff explicitly talks about "cdylib-only". That commit was part of https://github.com/rust-lang/miri/pull/1710 which fixed https://github.com/rust-lang/miri/issues/1705.

Reply:
https://github.com/rust-lang/miri/issues/1705 was about a crate that mixed cdylib and other crate types.

But still, a cdylib-only crate might be odd or useless, but it worked fine so far so I am concerned about breaking it as people might be relying on it.

Reply:
Yeah that's what I just wrote in the [2nd comment](https://github.com/rust-lang/miri/pull/4648#discussion_r2484756232). Hard to say what people rely on though.

What's the error one gets when having a cdylib-only crate?

Reply:
That error is kind of misleading since it's not about the target, it's about the codegen backend.